### PR TITLE
Rename Google.Cloud.Secrets.V1Beta1 to Google.Cloud.SecretManager.V1Beta

### LINKED ISSUE #3977: Synthesis failed for Google.Cloud.Secrets.V1Beta1
Hello! Autosynth couldn't regenerate Google.Cloud.Secrets.V1Beta1. :broken_heart:

Here's the output from running `synth.py`:

```
Cloning into 'working_repo'...
Switched to branch 'autosynth-Google.Cloud.Secrets.V1Beta1'
Running synthtool
['/tmpfs/src/git/autosynth/env/bin/python3', '-m', 'synthtool', 'synth.py', '--']
synthtool > Executing /tmpfs/src/git/autosynth/working_repo/apis/Google.Cloud.Secrets.V1Beta1/synth.py.
Cloning into 'gapic-generator-csharp'...
Submodule 'api-common-protos' (https://github.com/googleapis/api-common-protos.git) registered for path 'api-common-protos'
Submodule 'protobuf' (https://github.com/protocolbuffers/protobuf.git) registered for path 'protobuf'
Cloning into '/tmpfs/src/git/autosynth/working_repo/gapic-generator-csharp/api-common-protos'...
Cloning into '/tmpfs/src/git/autosynth/working_repo/gapic-generator-csharp/protobuf'...
Submodule path 'api-common-protos': checked out '93c9c383396b514193735292f6880fccc777614e'
Submodule path 'protobuf': checked out '815ff7e1fb2d417d5aebcbf5fc46e626b18dc834'
Submodule 'third_party/benchmark' (https://github.com/google/benchmark.git) registered for path 'protobuf/third_party/benchmark'
Submodule 'third_party/googletest' (https://github.com/google/googletest.git) registered for path 'protobuf/third_party/googletest'
Cloning into '/tmpfs/src/git/autosynth/working_repo/gapic-generator-csharp/protobuf/third_party/benchmark'...
Cloning into '/tmpfs/src/git/autosynth/working_repo/gapic-generator-csharp/protobuf/third_party/googletest'...
Submodule path 'protobuf/third_party/benchmark': checked out '5b7683f49e1e9223cf9927b24f6fd3d6bd82e3f8'
Submodule path 'protobuf/third_party/googletest': checked out '5ec7f0c4a113e2f18ac2c6cc7df51ad6afc24081'
Cloning into 'gapic-generator'...
Cloning into 'googleapis'...
> Task :buildSrc:compileJava NO-SOURCE
> Task :buildSrc:compileGroovy
> Task :buildSrc:processResources NO-SOURCE
> Task :buildSrc:classes
> Task :buildSrc:jar
> Task :buildSrc:assemble
> Task :buildSrc:compileTestJava NO-SOURCE
> Task :buildSrc:compileTestGroovy NO-SOURCE
> Task :buildSrc:processTestResources NO-SOURCE
> Task :buildSrc:testClasses UP-TO-DATE
> Task :buildSrc:test NO-SOURCE
> Task :buildSrc:check UP-TO-DATE
> Task :buildSrc:build
> Task :extractIncludeProto
> Task :extractProto
> Task :generateProto

> Task :compileJava
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.

> Task :processResources
> Task :createProperties
> Task :classes
> Task :shadowJar

BUILD SUCCESSFUL in 14s
7 actionable tasks: 7 executed
Building existing version of Google.Cloud.Secrets.V1Beta1 for compatibility checking
Generating Google.Cloud.Secrets.V1Beta1
API Google.Cloud.Secrets.V1Beta1 has broken namespace declarations
synthtool > Failed executing /bin/bash generateapis.sh --check_compatibility Google.Cloud.Secrets.V1Beta1:

None
synthtool > Wrote metadata to synth.metadata.
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.1/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.1/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/git/autosynth/env/lib/python3.6/site-packages/synthtool/__main__.py", line 109, in <module>
    main()
  File "/tmpfs/src/git/autosynth/env/lib/python3.6/site-packages/click/core.py", line 764, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/git/autosynth/env/lib/python3.6/site-packages/click/core.py", line 717, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/git/autosynth/env/lib/python3.6/site-packages/click/core.py", line 956, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/git/autosynth/env/lib/python3.6/site-packages/click/core.py", line 555, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/git/autosynth/env/lib/python3.6/site-packages/synthtool/__main__.py", line 96, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 205, in _call_with_frames_removed
  File "/tmpfs/src/git/autosynth/working_repo/apis/Google.Cloud.Secrets.V1Beta1/synth.py", line 20, in <module>
    hide_output = False)
  File "/tmpfs/src/git/autosynth/env/lib/python3.6/site-packages/synthtool/shell.py", line 39, in run
    raise exc
  File "/tmpfs/src/git/autosynth/env/lib/python3.6/site-packages/synthtool/shell.py", line 33, in run
    encoding="utf-8",
  File "/home/kbuilder/.pyenv/versions/3.6.1/lib/python3.6/subprocess.py", line 418, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '('/bin/bash', 'generateapis.sh', '--check_compatibility', 'Google.Cloud.Secrets.V1Beta1')' returned non-zero exit status 1.

Synthesis failed

```

Google internal developers can see the full log [here](https://sponge/315af26d-1ec9-4d8c-bf3e-bc6d632732c9).

### PR DESCRIPTION
**Pull request by jskeet**
Fixes #3977.

We'll do a release in January.

### CODE REVIEW

File: apis/apis.json [hunk 1]
```diff
@@ -740,13 +740,13 @@
   },
 
   {
-    "id": "Google.Cloud.Secrets.V1Beta1",
+    "id": "Google.Cloud.SecretManager.V1Beta1",
     "generator": "micro",
     "protoPath": "google/cloud/secrets/v1beta1",
```
Comment (amanda-tarafa) [inline apis/apis.json]: And the proto path hasn't changed?

**Comment by amanda-tarafa** [comment]
And I think the PR will need to be splitten in two, first all the changes, and then apis.json and regenerate projects? I think that's what's making CI fail.

File: apis/apis.json (hunk 1 above)
Comment (jskeet) [reply apis/apis.json]: Proto path hasn't changed. Will fix the readme in another PR later.

Comment (jskeet) [reply apis/apis.json]: Proto path hasn't changed. Will fix the readme in another PR later.

**Comment by jskeet** [comment]
We can ignore CI - it always fails when things are deleted.

**Comment by jskeet** [comment]
Oh, but maybe not Travis. Will sort it all out tomorrow probably.

**Comment by jskeet** [comment]
Okay, I believe both CI failures *are* due to it still trying to build Secrets. Merging.

**Pull request merged**